initfiles/sbin/hpcc_setenv.in

```diff
@@ -86,9 +86,20 @@ function pidwait_fn () {
 platform='unknown'
 unamestr=`uname`
 if [[ "$unamestr" == 'Linux' ]]; then
-   platform='linux'
+    platform='linux'
+    which nproc >/dev/null 2>&1
+    rc=$?
+    if [[ $rc -eq 0 ]] ; then
+       num_cpus=$(nproc)
+    else
+       num_cpus=$(grep -c ^processor /proc/cpuinfo 2>/dev/null)
```

Comment:
From my reading, the danger of /proc/cpuinfo in some (not all) containers is that if it doesn't have it's own isolated kernel, it will see all the processors, and not just what's available to that specific container.  Nproc should work fine, as it will only return the number of processors it sees available within that container (unless --all is specified).